### ISSUE TITLE
Allow AMQP arguments in AMQPQueueDeclare instead of always passing nil (Enables Quorum Queues and other functionality)

### ISSUE THREAD

**Issue by jlee-idbydna** [argoproj/argo-events#1479]
**Is your feature request related to a problem? Please describe.**
We would like to use argo-events to listen to our ingress queue which is a RabbitMQ quorum queue.  However, doing so requires the x-queue-type argument to be set upon queue declaration.  https://www.rabbitmq.com/quorum-queues.html#declaring

**Describe the solution you'd like**
A simple solution may be to allow a key/value map in AMQPQueueDeclareConfig for the AMQP arguments, which can be passed through to amqlib.QueueDeclare instead of the hard coded nil value: https://github.com/argoproj/argo-events/blob/f85b8d01b631b9ae92e5e16435e9c4a50354455a/eventsources/sources/amqp/start.go#L257

**Describe alternatives you've considered**
There are no reasonable alternatives for us without introducing another queue between the quorum queue and argo-events, which defeats the resiliency goal of the quorum queue

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by pdaBlueVoyant** [comment]
Funny this is here, I was about to write the exact same request.  To meet our goals I may have to create a local version unless this can get added in the next 3 weeks.  If we cannot find a work around and I add this I'll create a PR.

**Comment by Capable-Disk4147** [comment]
I forked the repo to start this work for @pdaBlueVoyant (he's my architect).

I know Go well, but I haven't used much protobuf, Argo (either workflow or events), or RabbitMQ so while the code change is probably small, I'm not sure on some of the things I'm seeing. I didn't write tests yet because it's only a couple lines of code, and my entire premise might be off. Therefore, I'm opening a PR to make some questions.

**Comment by whynowy** [comment]
Any comments on the spec like following:

```yaml
     exchangeName: test
     exchangeType: fanout
     routingKey: hello
     queueDeclare:
       name: xxx
       durable: xxxxx
       arguments: |-
         a: b
         c:
           d: e
           f: g
```

Where `arguments` is a string in yaml format. @jlee-idbydna, @pdaBlueVoyant

**Comment by Capable-Disk4147** [comment]
hm, I don't entirely know what that Yaml syntax means until I test things today. Are you saying you suspect that it is all one escaped string?

**Comment by whynowy** [comment]
> hm, I don't entirely know what that Yaml syntax means until I test things today. Are you saying you suspect that it is all one escaped string? Edit: learning and testing the whole system/process today and/or tomorrow to get closer to where I can make intelligent statements on this

https://www.baeldung.com/yaml-multi-line

**Comment by Capable-Disk4147** [comment]
I see. Thanks. I'm still working on the whole environment setup, so it might be tomorrow before I test, but I will push the change from bytes to string today

**Comment by Capable-Disk4147** [comment]
Yes, it can be. It's merged and went into production in our company with a large load using it. Since then we've actually. migrated off argo events, so I am glad it's still working! I cannot close this since I didn't open it.

@jlee-idbydna or @whynowy Could one of you close this?
